examples/updated-vm-with-contracts/actions/transfer_test.go

```diff
@@ -0,0 +1,164 @@
+	"github.com/ava-labs/hypersdk/examples/morpheusvm/storage"
+	"github.com/ava-labs/hypersdk/state"
+)
+
+func TestTransferAction(t *testing.T) {
+	addr := codectest.NewRandomAddress()
+
+	tests := []chaintest.ActionTest{
+		{
+			Name:  "ZeroTransfer",
+			Actor: codec.EmptyAddress,
+			Action: &Transfer{
+				To:    codec.EmptyAddress,
+				Value: 0,
+			},
+			ExpectedErr: ErrOutputValueZero,
+		},
+		{
+			Name:  "InvalidAddress",
+			Actor: codec.EmptyAddress,
+			Action: &Transfer{
+				To:    codec.EmptyAddress,
+				Value: 1,
+			},
+			State:       chaintest.NewInMemoryStore(),
+			ExpectedErr: storage.ErrInvalidAddress,
+		},
+		{
+			Name:  "NotEnoughBalance",
+			Actor: codec.EmptyAddress,
+			Action: &Transfer{
+				To:    codec.EmptyAddress,
+				Value: 1,
+			},
+			State: func() state.Mutable {
+				s := chaintest.NewInMemoryStore()
+				_, err := storage.AddBalance(
+					context.Background(),
+					s,
+					codec.EmptyAddress,
+					0,
+					true,
+				)
+				require.NoError(t, err)
+				return s
+			}(),
+			ExpectedErr: storage.ErrInvalidBalance,
+		},
+		{
+			Name:  "SelfTransfer",
+			Actor: codec.EmptyAddress,
+			Action: &Transfer{
+				To:    codec.EmptyAddress,
+				Value: 1,
+			},
+			State: func() state.Mutable {
+				store := chaintest.NewInMemoryStore()
+				require.NoError(t, storage.SetBalance(context.Background(), store, codec.EmptyAddress, 1))
+				return store
+			}(),
+			Assertion: func(ctx context.Context, t *testing.T, store state.Mutable) {
+				balance, err := storage.GetBalance(ctx, store, codec.EmptyAddress)
+				require.NoError(t, err)
+				require.Equal(t, balance, uint64(1))
+			},
+			ExpectedOutputs: &TransferResult{
+				SenderBalance:   0,
+				ReceiverBalance: 1,
+			},
+		},
+		{
+			Name:  "OverflowBalance",
+			Actor: codec.EmptyAddress,
+			Action: &Transfer{
+				To:    codec.EmptyAddress,
+				Value: math.MaxUint64,
+			},
+			State: func() state.Mutable {
+				store := chaintest.NewInMemoryStore()
+				require.NoError(t, storage.SetBalance(context.Background(), store, codec.EmptyAddress, 1))
+				return store
+			}(),
+			ExpectedErr: storage.ErrInvalidBalance,
+		},
+		{
+			Name:  "SimpleTransfer",
+			Actor: codec.EmptyAddress,
+			Action: &Transfer{
+				To:    addr,
+				Value: 1,
+			},
+			State: func() state.Mutable {
+				store := chaintest.NewInMemoryStore()
+				require.NoError(t, storage.SetBalance(context.Background(), store, codec.EmptyAddress, 1))
+				return store
+			}(),
+			Assertion: func(ctx context.Context, t *testing.T, store state.Mutable) {
+				receiverBalance, err := storage.GetBalance(ctx, store, addr)
+				require.NoError(t, err)
+				require.Equal(t, receiverBalance, uint64(1))
+				senderBalance, err := storage.GetBalance(ctx, store, codec.EmptyAddress)
+				require.NoError(t, err)
+				require.Equal(t, senderBalance, uint64(0))
+			},
+			ExpectedOutputs: &TransferResult{
+				SenderBalance:   0,
+				ReceiverBalance: 1,
+			},
+		},
+	}
```

Comment:
🤔 I don't know about Go best practices, but at the very least, it might make sense to have something like:
`transfer_test.go` - unit tests
`transfer_integration_test.go` - integration tests

Or maybe they should live in a level above the package? If you're testing `action` and `storage`, maybe the integration tests live in the parent directory of both packages?